Lib/idlelib/idle_test/test_config.py

```diff
@@ -185,6 +186,358 @@ def test_save(self):
             self.assertFalse(os.path.exists(path))
 
 
+class IdleConfTest(unittest.TestCase):
+    """Test for idleConf"""
+
+    def new_config(self, _utest=False):
+        return config.IdleConf(_utest=_utest)
+
+    def mock_config(self):
+        """Return a mocked idleConf
+
+        Both default and user config used the same config-*.def
+        """
+        import sys
+        conf = config.IdleConf(_utest=True)
+        if __name__ != '__main__':
+            idle_dir = os.path.dirname(__file__)
+        else:
+            idle_dir = os.path.abspath(sys.path[0])
+        for ctype in conf.config_types:
+            config_path = os.path.join(idle_dir, '../config-%s.def' % ctype)
+            conf.defaultCfg[ctype] = config.IdleConfParser(config_path)
+            conf.userCfg[ctype] = config.IdleUserConfParser(config_path)
+        conf.LoadCfgFiles()
+
+        return conf
```

Comment:
The point of this is unclear just from reading.  Having duplicate info in defaultCfg and userCfg is contrary to intended reality (though users can over-ride by hand editing).  In general, I think userCfg should be loaded with fake data with read_string the way I did in the CurrentColorKeysTest.

 Re-reading and parsing the .def files twice for each of multiple functions is needlessly slow.  The files can be read once and then loaded into empty parsers.  Besides being faster for reloads, not passing the actual file name prevents accidentally writing the files.  I believe that idleConf never changes defaultCfg.  (If true, we could in a followup isssue define a readonly IdleDefConfParser that raised on any attempt to change defaults.)  If we want to change a particular default parser for a test, we should read into a particular parser in an expanded testCfg. 

I did not yet make any changes based on the above.